framework/src/base/FEProblemBase.C

```diff
@@ -3590,6 +3590,8 @@ FEProblemBase::computeResidualType(const NumericVector<Number>& soln, NumericVec
 
   executeControls(EXEC_LINEAR);
 
+  _current_execute_on_flag = EXEC_NONE;
+
   _app.getOutputWarehouse().residualSetup();
 
   _nl->computeResidual(residual, type);
```

Comment:
What about this line? You're setting the flag back before we call compute residual.

Reply:
I dont see why we are unsetting it though ?

This is causing issues for the execution order printing, as the flag is wrong when we print it.